mistralrs-core/src/models/quantized_llama.rs

```diff
@@ -396,6 +422,9 @@ impl ModelWeights {
         start_offsets_kernel: Tensor,
         context_lens: Vec<(usize, usize)>,
     ) -> Result<Tensor> {
+        let (bz, seq_len, _) = x.dims3()?;
+        let via_f16 = bz * seq_len > 256;
```

Comment:
I'm not sure if this is optimal. Benchmarking and tuning is still required.

Reply:
```suggestion
        let (bz, seq_len, _) = x.dims3()?;
        let via_f16 = seq_len > 32;
```